### PR TITLE
Update simple pipeline example

### PR DESCRIPTION
**Pull request by objorkman**
Changes:
* Update `simple_pipeline.py` for erdos redesign
* Fix multiprocessing process issue on macOS for Python 3.8+

### CODE REVIEW

File: python/examples/simple_pipeline.py [hunk 1]
```diff
@@ -7,75 +7,67 @@
...
 
-    def run(self):
+    def run(self, read_stream):
         while True:
-            data = self.read_stream.read()
+            data = read_stream.read()
             print("PullOp: received {data}".format(data=data))
 
+    def on_data(self, context, data):
+        pass
```
Comment (pschafhalter) [inline python/examples/simple_pipeline.py:48]: Could you delete this function?

File: python/examples/simple_pipeline.py [hunk 2]
```diff
@@ -7,75 +7,67 @@
...
-    def run(self):
+    def run(self, read_stream):
         while True:
-            data = self.read_stream.try_read()
+            data = read_stream.try_read()
             print("TryPullOp: received {data}".format(data=data))
             time.sleep(0.5)
 
+    def on_data(self, context, data):
+        pass
```
Comment (pschafhalter) [inline python/examples/simple_pipeline.py:62]: Also delete this function

**Comment by pschafhalter** [comment]
Could you fix the python formatting?

**Review by pschafhalter — APPROVED** [review]
LGTM!

File: python/examples/simple_pipeline.py [hunk 3]
```diff
@@ -7,75 +7,65 @@
...
-        read_stream.add_callback(CallbackOp.callback)
+class CallbackOp(Sink):
+    def __init__(self):
+        print("initializing callback op")
 
-    @staticmethod
-    def callback(msg):
-        print("CallbackOp: received {msg}".format(msg=msg))
+    def run(self, read_stream):
+        print("Executing run with the ReadStream {}".format(read_stream))
```
Comment (sukritkalra) [inline python/examples/simple_pipeline.py:33]: Can you please remove this? This example is supposed to show only the callback-driven execution. `PullOp` and `TryPullOp` exemplify a `run`-driven execution.

**Review by sukritkalra — APPROVED** [review]
LGTM. Please see a minor comment.

**Pull request merged**